packages/react-refresh/src/ReactFreshRuntime.js

```diff
@@ -472,7 +472,8 @@ export function injectIntoGlobalHook(globalObject: any): void {
     // Do the same for any already injected roots.
     // This is useful if ReactDOM has already been initialized.
     // https://github.com/facebook/react/issues/17626
-    hook.renderers.forEach((injected, id) => {
+    const renderers = hook.renderers || hook._renderers; // DevTools v4 || v3
+    renderers.forEach((injected, id) => {
```

Comment:
In v4, [`renderers` is a `Map`](https://github.com/facebook/react/blob/1e1a98942225771f86d8a81b5ac561a9fbff9263/packages/react-devtools-shared/src/hook.js#L275) but in v3 [`_renderers` is an `Object`](https://github.com/facebook/react-devtools/blob/d839081f79f681617df7584a8ffd2f4163fa40b9/backend/installGlobalHook.js#L127). So `.forEach()` wouldn't work on v3. You'd need to do a different loop for v3.

Reply:
I admire your diligence.

Reply:
![image](https://user-images.githubusercontent.com/29597/72390300-3ac7c900-36df-11ea-8559-bd1d1a4d43dc.png)